cuda_pathfinder/cuda/pathfinder/_dynamic_libs/find_nvidia_dynamic_lib.py

```diff
@@ -146,53 +158,57 @@ def __init__(self, libname: str):
         self.libname = libname
         self.error_messages: list[str] = []
         self.attachments: list[str] = []
-        self.abs_path = None
+        self.abs_path: Optional[str] = None
 
+    def try_site_packages(self) -> None:
         if IS_WINDOWS:
-            self.lib_searched_for = f"{libname}*.dll"
+            self.lib_searched_for = f"{self.libname}*.dll"
             if self.abs_path is None:
```

Comment:
For consistency let's move all `self.abs_path` checks to the call site.

Reply:
Or maybe make all methods of this class return `abs_path` and not holding it as an internal state of this class.

Reply:
Done: commit ab2a349133216753124de80fb7b30f24b8307cb4